File: tireguide/src/main/java/com/droidsmith/tireguide/calcengine/Calculator.java
package com.droidsmith.tireguide.calcengine;

import android.text.TextUtils;

/**
 * Created by steve on 12/28/15. Calculates the proper tire pressure for a bicycle tire based on
 * weight and width of tire.
 */
public class Calculator {
	private double loadWeight;
	private String tireWidth;

	public Calculator(double loadWeight, String tireWidth) {
		this.loadWeight = loadWeight;
		this.tireWidth = tireWidth;
	}

	/**
	 * Uses magical numbers to determine the optimal tire pressure;
	 * @return The magical tire pressure;
	 */
	public double psi() {
		double m = 0.0;
		double b = 0.0;
		double psi;

		if (TextUtils.equals("20", tireWidth)) {
			m = 1.363636364f;
			b = -11f;
		} else if (TextUtils.equals("21", tireWidth)) {
			m = 1.181818182f;
			b = 0f;
		} else if (TextUtils.equals("22", tireWidth)) {
			m = 1.169601449f;
			b = -6.260144928f;
		} else if (TextUtils.equals("23", tireWidth)) {
			m = 1.136363636f;
			b = -9.5f;
		} else if (TextUtils.equals("24", tireWidth)) {
			m = 1.053507905f;
			b = -10.26594203f;
		} else if (TextUtils.equals("25", tireWidth)) {
			m = 1f;
			b = -13f;
		} else if (TextUtils.equals("26", tireWidth)) {
			m = 0.9522727273f;
			b = -12.55f;
		} else if (TextUtils.equals("27", tireWidth)) {
			m = 0.886363636f;
			b = -8.5f;
		} else if (TextUtils.equals("28", tireWidth)) {
			m = .8022580702f;
			b = -6.348534297f;
		}

		psi = loadWeight * m + b;
		return psi;
	}
}

File: tireguide/src/main/java/com/droidsmith/tireguide/calcengine/Calculator.kt
package com.droidsmith.tireguide.calcengine

import java.util.Objects

/**
 * Calculates the proper tire pressure for a bicycle tire based on weight and width of tire.
 */
class Calculator {

	/**
	 * Uses magical numbers to determine the optimal tire pressure.
	 * @param loadWeight The weight in pounds exerted on the wheel.
	 * @param tireWidth The width of the tire.
	 * @return The magical tire pressure using formula y=mx+b.
	 */
	fun psi(loadWeight: Double, tireWidth: String): Double {
		var m = 0.0
		var b = 0.0
		val psi: Double

		when {
			Objects.equals(tireWidth, "20") -> {
				m = 1.363636364
				b = -11.0
			}
			Objects.equals(tireWidth, "21") -> {
				m = 1.181818182
				b = 0.0
			}
			Objects.equals(tireWidth, "22") -> {
				m = 1.169601449
				b = -6.260144928
			}
			Objects.equals(tireWidth, "23") -> {
				m = 1.136363636
				b = -9.5
			}
			Objects.equals(tireWidth, "24") -> {
				m = 1.053507905
				b = -10.26594203
			}
			Objects.equals(tireWidth, "25") -> {
				m = 1.0
				b = -13.0
			}
			Objects.equals(tireWidth, "26") -> {
				m = 0.9522727273
				b = -12.55
			}
			Objects.equals(tireWidth, "27") -> {
				m = 0.886363636
				b = -8.5
			}
			Objects.equals(tireWidth, "28") -> {
				m = .8022580702
				b = -6.348534297
			}
		// Find the slope of y=mx+b
		}

		psi = loadWeight * m + b // Find the slope of y=mx+b
		return Math.round(psi).toDouble()
	}
}
